File: LeetCode/1800最大升序子数组和.cpp
int maxAscendingSum(int* nums, int numsSize){
    int max = INT_MIN, sum = 0;
    for (int i = 0; i < numsSize; i++) {
        sum = (i != 0 && nums[i] > nums[i - 1]) ? sum + nums[i] : nums[i];
        max = fmax(max, sum);
    }
    return max;
}

File: LeetCode/1800最大升序子数组和.go
func maxAscendingSum(nums []int) int {
	var ans, sum = 0, 0
	for i := 0; i < len(nums); i++ {
		if i != 0 && nums[i-1] < nums[i] {
			sum += nums[i]
		} else {
			sum = nums[i]
		}
		ans = int(math.Max(float64(ans), float64(sum)))
	}
	return ans
}